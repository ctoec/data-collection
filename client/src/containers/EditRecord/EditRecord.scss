@import '~@ctoec/component-library/dist/assets/styles/index.scss';

.oec-tab-nav--tab {
  &:not(&__active) {
    .svg-gold-20v {
      svg {
        @extend .oec-inline-icon--incomplete;
      }
    }
  }

  svg, path {
    fill: currentColor;
  }
}
