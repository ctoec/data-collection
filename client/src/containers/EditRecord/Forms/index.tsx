export * from './EnrollmentFunding/Form';
export * from './FamilyInfo/Form';
export * from './CareForKids';
export * from './ChildInfo/Form';
export * from './FamilyIncome/Form';
