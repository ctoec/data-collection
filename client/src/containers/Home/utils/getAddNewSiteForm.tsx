import { TextInput } from '@ctoec/component-library';
import React from 'react';
import { AddSiteRequest } from '../../../shared/payloads/AddSiteRequest';

/**
 * Function to expand a new set of fields for a user to add a new
 * site in the revision request form. The form tracks which site
 * to update among the state list sites by the index in the list.
 * @param addRequest
 * @param idx
 * @param setAddRequests
 */
export const getAddNewSiteForm = (
  addRequest: AddSiteRequest,
  idx: number,
  setAddRequests: React.Dispatch<React.SetStateAction<AddSiteRequest[]>>
) => {
  return (
    <>
      <p className="text-bold">New site information</p>
      <TextInput
        label="New site name"
        id="new-site-name"
        type="input"
        defaultValue={addRequest.siteName}
        onChange={(e) => {
          const change = e.target.value;
          setAddRequests((o: AddSiteRequest[]) => {
            o[idx].siteName = change;
            return o;
          });
          return change;
        }}
      />
      <TextInput
        label="License ID"
        id="new-site-license"
        type="input"
        defaultValue={addRequest.licenseId}
        onChange={(e) => {
          const change = e.target.value;
          setAddRequests((o: AddSiteRequest[]) => {
            o[idx].licenseId = change;
            return o;
          });
          return change;
        }}
      />
      <TextInput
        label="NAEYC ID"
        id="new-site-naeyc"
        type="input"
        defaultValue={addRequest.naeycId}
        onChange={(e) => {
          const change = e.target.value;
          setAddRequests((o: AddSiteRequest[]) => {
            o[idx].naeycId = change;
            return o;
          });
          return change;
        }}
      />
      <TextInput
        className="margin-bottom-4"
        label="Registry ID"
        id="new-site-registry"
        type="input"
        defaultValue={addRequest.registryId}
        onChange={(e) => {
          const change = e.target.value;
          setAddRequests((o: AddSiteRequest[]) => {
            o[idx].registryId = change;
            return o;
          });
          return change;
        }}
      />
    </>
  );
};
