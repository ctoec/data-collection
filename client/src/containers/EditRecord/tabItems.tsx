import {
  Info,
  TextWithIcon,
  TextWithIconProps,
} from '@ctoec/component-library';
import React from 'react';
import {
  ChildInfoForm,
  ChildIdentifiersForm,
  FamilyAddressForm,
} from '../../components/Forms';
import { EnrollmentFundingForm } from './EnrollmentFunding/Form';
import { FamilyIncomeForm } from './FamilyIncome/Form';
import {
  SECTION_KEYS,
  formSections,
} from '../../components/Forms/formSections';
import { RecordFormProps } from '../../components/Forms/types';

const commonTextWithIconProps: Omit<TextWithIconProps, 'text'> = {
  Icon: Info,
  iconSide: 'right',
  className: 'svg-gold-20v',
};

export const editForms = [
  { key: SECTION_KEYS.IDENT, form: ChildIdentifiersForm },
  { key: SECTION_KEYS.DEMO, form: ChildInfoForm },
  { key: SECTION_KEYS.FAMILY, form: FamilyAddressForm },
  { key: SECTION_KEYS.INCOME, form: FamilyIncomeForm },
  { key: SECTION_KEYS.ENROLLMENT, form: EnrollmentFundingForm },
];

export const getTabItems = (commonFormProps: RecordFormProps) =>
  formSections.map(({ key, name, hasError }) => {
    return {
      id: key,
      tabText: name,
      tabTextFormatter: (_name: string) =>
        hasError(commonFormProps.child) ? (
          <TextWithIcon {...commonTextWithIconProps} text={_name} />
        ) : (
          _name
        ),
    };
  });

export const getTabContent = (
  activeTab: string,
  commonFormProps: RecordFormProps
) => {
  const EditForm =
    editForms.find((e) => e.key === activeTab)?.form || (() => <></>);
  return <EditForm {...commonFormProps} />;
};
