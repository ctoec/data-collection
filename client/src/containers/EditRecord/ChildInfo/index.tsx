import React, { useContext, useState } from 'react';
import {
  SasidField,
  FirstNameField,
  MiddleNameField,
  LastNameField,
  SuffixField,
  DateOfBirthField,
  BirthCertificateFormFieldSet,
  RaceField,
  EthnicityField,
  GenderField,
  FosterCheckbox
} from './Fields';
import { Form, FormSubmitButton } from '@ctoec/component-library';
import AuthenticationContext from '../../../contexts/AuthenticationContext/AuthenticationContext';
import { apiPut } from '../../../utils/api';
import { Child } from '../../../shared/models';
// import { useFocusFirstError } from '../../../hooks/useFocusFirstError';

export default ({
  child,
  refetchChild,
}: {
  child: Child;
  refetchChild: () => void;
}) => {
  const { accessToken } = useContext(AuthenticationContext);
  const [saving, setSaving] = useState(false);
  // TODO: HOW ARE WE HANDLING ERRORS?
  // const [error, setError] = useState(null);

  // Focus should automatically be on first error on page
  // useFocusFirstError([error]);

  const onFormSubmit = (_child: Child) => {
    setSaving(true);
    apiPut(`children/${child.id}`, _child, {
      accessToken,
      jsonParse: false
    })
      .then(refetchChild)
      .catch((err) => {
        // TODO: HOW ARE WE HANDLING ERRORS?  Do validation errors come back with the child?
        console.error(`There was an error saving child info: ${err}`);
      })
      .finally(() => setSaving(false));
  };

  if (!child) {
    throw new Error('Child info rendered without child');
  }

  return (
    <Form<Child>
      className="ChildInfoForm usa-form"
      data={child}
      onSubmit={onFormSubmit}
      noValidate
      autoComplete="off"
    >
      <div className="grid-row grid-gap">
        <div className="mobile-lg:grid-col-12">
          <SasidField />
        </div>
        <div className="mobile-lg:grid-col-9">
          <FirstNameField />
        </div>
        <div className="mobile-lg:grid-col-9">
          <MiddleNameField />
        </div>
        <div className="display-flex flex-row flex-align-end grid-row grid-gap">
          <div className="mobile-lg:grid-col-9">
            <LastNameField />
          </div>
          <div className="mobile-lg:grid-col-3">
            <SuffixField />
          </div>
        </div>
      </div>
      <h2>Date of birth</h2>
      <DateOfBirthField />
      <h2>Birth certificate</h2>
      <BirthCertificateFormFieldSet />
      <h2>Race</h2>
      <RaceField />
      <h2>Ethnicity</h2>
      <EthnicityField />
      <h2>Gender</h2>
      <GenderField />
      <h2>Foster</h2>
      <FosterCheckbox />
      <FormSubmitButton text={saving ? 'Saving...' : 'Save'} disabled={saving} />
    </Form>
  );
};
