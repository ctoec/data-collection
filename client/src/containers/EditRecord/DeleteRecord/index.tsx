import React, { useContext, useState } from 'react';
import { useHistory } from 'react-router-dom';
import AuthenticationContext from '../../../contexts/AuthenticationContext/AuthenticationContext';
import { apiDelete } from '../../../utils/api';
import { Child } from '../../../shared/models';
import { Button } from '@ctoec/component-library';

type DeleteProps = {
  child: Child;
  toggleOpen: () => void;
};

export const DeleteRecord: React.FC<DeleteProps> = ({ child, toggleOpen }) => {
  const { accessToken } = useContext(AuthenticationContext);
  const history = useHistory();

  const [isDeleting, setIsDeleting] = useState(false);

  function deleteRecord() {
    setIsDeleting(true);
    apiDelete(`children/${child.id}`, { accessToken })
      // TODO: Swap this total hack out for the roster page
      // once we have that implemented
      .then(() => history.push('/check-data/1'))

      .catch((err) => {
        console.log(err);
      })
      .finally(() => {
        toggleOpen();
        setIsDeleting(false);
      });
  }

  return (
    <div className="grid-container">
      <div className="grid-row margin-top-2">
        <h2>
          Do you want to delete the enrollment for {child.firstName}{' '}
          {child.lastName}?
        </h2>
      </div>
      <div className="grid-row margin-top-2">
        <span>
          Deleting an enrollment record will permanently remove all of its data
        </span>
      </div>
      <div className="margin-top-4">
        <div className="grid-row flex-first-baseline space-between-4">
          <Button appearance="outline" onClick={toggleOpen} text="No, cancel" />
          <Button
            appearance={isDeleting ? 'outline' : 'default'}
            onClick={deleteRecord}
            text={isDeleting ? 'Deleting record...' : 'Yes, delete record'}
          />
        </div>
      </div>
    </div>
  );
};
