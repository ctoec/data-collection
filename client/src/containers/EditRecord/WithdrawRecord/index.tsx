import React, { useContext, useState } from 'react';
import {
  Tag,
  Form,
  Button,
  FormSubmitButton,
  Alert,
} from '@ctoec/component-library';
import { Enrollment, ReportingPeriod } from '../../../shared/models';
import { apiPost } from '../../../utils/api';
import AuthenticationContext from '../../../contexts/AuthenticationContext/AuthenticationContext';
import { useHistory } from 'react-router-dom';
import {
  EnrollmentEndDateField,
  ReportingPeriodField,
} from '../../../components/EditForms/EnrollmentFunding/Fields';
import { ExitReasonField } from './Fields/ExitReason';
import { Withdraw } from '../../../shared/payloads';

type WithdrawProps = {
  childName: string;
  enrollment: Enrollment;
  reportingPeriods: ReportingPeriod[];
  toggleOpen: () => void;
};
export const WithdrawRecord: React.FC<WithdrawProps> = ({
  childName,
  enrollment,
  reportingPeriods,
  toggleOpen,
}) => {
  const [isSaving, setIsSaving] = useState(false);
  const [error, setError] = useState<string>();
  const { accessToken } = useContext(AuthenticationContext);
  const history = useHistory();

  const onSubmit = (withdraw: Withdraw) => {
    setIsSaving(true);
    apiPost(`enrollments/${enrollment.id}/withdraw`, withdraw, {
      accessToken,
      jsonParse: false,
    })
      .then(() => {
        history.push('/roster', {
          alerts: [
            {
              type: 'success',
              heading: 'Record withdrawn',
              text: `${childName} has been withdrawn from your program`,
            },
          ],
        });
      })
      .catch((err) => {
        console.log(err);
        setError(`Unable to withdraw ${childName}`);
      })
      .finally(() => setIsSaving(false));
  };

  const activeFunding = (enrollment.fundings || []).find(
    (f) => !f.lastReportingPeriod
  );
  return (
    <div className="grid-container">
      {!!error && <Alert type="error" text={error} />}
      <h1>Withdraw {childName}</h1>
      <div className="grid-row">
        <div className="grid-col">
          <p>{enrollment.site.name}</p>
          <p>Age: {enrollment.ageGroup}</p>
          <p>Enrollment date: {enrollment.entry?.format('MM/DD/YYYY')}</p>
        </div>
        <div className="grid-col">
          {activeFunding && (
            <>
              <p>
                <Tag text={activeFunding.fundingSpace.source} />
              </p>
              <p>Contract space: {activeFunding.fundingSpace.time}</p>
              <p>
                First reporting period:{' '}
                {activeFunding.firstReportingPeriod.period.format('MMMM YYYY')}
              </p>
            </>
          )}
        </div>
      </div>
      <Form<Withdraw>
        onSubmit={onSubmit}
        data={{} as Withdraw}
        className="usa-form"
      >
        <EnrollmentEndDateField<Withdraw>
          accessor={(data) => data.at('exitDate')}
        />
        <ExitReasonField />
        {!!activeFunding && (
          <ReportingPeriodField<Withdraw>
            reportingPeriods={reportingPeriods.filter(
              (rp) => rp.type === activeFunding.fundingSpace.source
            )}
            isLast={true}
            accessor={(data) => data.at('funding').at('lastReportingPeriod')}
          />
        )}
        <Button appearance="outline" text="Cancel" onClick={toggleOpen} />
        <FormSubmitButton text={isSaving ? 'Withdrawing...' : 'Withdraw'} />
      </Form>
    </div>
  );
};
