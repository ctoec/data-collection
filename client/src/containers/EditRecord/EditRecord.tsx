import React, { useState, useContext, useEffect } from 'react';
import { useParams, useLocation, useHistory } from 'react-router-dom';
import {
  AlertProps,
  ErrorBoundary,
  LoadingWrapper,
  TabNav,
} from '@ctoec/component-library';
import AuthenticationContext from '../../contexts/AuthenticationContext/AuthenticationContext';
import { apiGet } from '../../utils/api';
import { BackButton } from '../../components/BackButton';
import { WithdrawRecord } from './WithdrawRecord/WithdrawRecord';
import { DeleteRecord } from './DeleteRecord/DeleteRecord';
import { useAlerts } from '../../hooks/useAlerts';
import { getH1RefForTitle } from '../../utils/getH1RefForTitle';
import { useFocusFirstError } from '../../hooks/useFocusFirstError';
import { getTabItems, getTabContent } from './tabItems';
import {
  SECTION_KEYS,
  formSections,
} from '../../components/Forms/formSections';
import { Child } from '../../shared/models';
import { defaultErrorBoundaryProps } from '../../utils/defaultErrorBoundaryProps';

const EditRecord: React.FC = () => {
  const h1Ref = getH1RefForTitle('Edit record');
  const { childId } = useParams() as {
    childId: string;
  };
  const { accessToken } = useContext(AuthenticationContext);
  const { alertElements, setAlerts } = useAlerts();
  const [child, setChild] = useState<Child>();

  // Persist active tab in URL hash
  const activeTab = useLocation().hash.slice(1);
  // Clear any previously displayed alerts from other tabs
  useEffect(() => {
    setAlerts([]);
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [activeTab]);
  const history = useHistory();
  // and make child tab active by default if no hash
  // (but only on first render)
  useEffect(() => {
    if (!activeTab) {
      history.replace({ hash: SECTION_KEYS.IDENT });
    }
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);

  // Child re-fetch
  const [triggerRefetchCounter, setTriggerRefetchCounter] = useState(0);
  useEffect(() => {
    apiGet(`children/${childId}`, accessToken)
      .then((updatedChild) => {
        setChild(updatedChild);

        const newAlerts: AlertProps[] = [];
        const missingInfoAlertProps = getMissingInfoAlertProps(updatedChild);
        // Always set missing info alert, if one exists
        if (missingInfoAlertProps) {
          newAlerts.push(missingInfoAlertProps);
        }

        // Only set success alert on a GET that happens after an update (refetch count > 0)
        if (triggerRefetchCounter > 0) {
          newAlerts.push({
            type: 'success',
            heading: 'Record updated',
            text: `Your changes to ${updatedChild?.firstName} ${updatedChild?.lastName}'s record have been saved.`,
          });
        }
        setAlerts(newAlerts);
      })
      .catch((err) => {
        throw new Error(err);
      });
  }, [accessToken, childId, triggerRefetchCounter]);

  useFocusFirstError([child]);

  const commonFormProps = {
    child,
    afterSaveSuccess: () => setTriggerRefetchCounter((r) => r + 1),
    setAlerts,
  };
  const activeEnrollment = (child?.enrollments || []).find((e) => !e.exit);
  return (
    <div className="grid-container">
      <BackButton />
      {alertElements}
      <div className="display-flex flex-justify">
        <div>
          <h1 ref={h1Ref} className="margin-top-0">
            <span className="h2 h2--lighter">Edit record </span>
            {child ? `${child.firstName} ${child.lastName}` : 'Loading...'}
          </h1>
        </div>
        {child && (
          <div className="display-flex flex-col flex-align-center">
            {!!activeEnrollment && (
              <>
                <WithdrawRecord child={child} enrollment={activeEnrollment} />
              </>
            )}
            <DeleteRecord child={child} setAlerts={setAlerts} />
          </div>
        )}
      </div>
      <div className="grid-col">
        <LoadingWrapper loading={!child}>
          <TabNav
            items={getTabItems(commonFormProps)}
            activeId={activeTab}
            onClick={(tabId) => {
              history.replace({ hash: tabId });
            }}
          >
            <ErrorBoundary alertProps={{ ...defaultErrorBoundaryProps }}>
              {child ? getTabContent(activeTab, commonFormProps) : <></>}
            </ErrorBoundary>
          </TabNav>
        </LoadingWrapper>
      </div>
    </div>
  );
};

const getMissingInfoAlertProps: (child: Child) => AlertProps | undefined = (
  child: Child
) => {
  const formsWithErrors = formSections.filter((section) =>
    section.hasError(child)
  );
  if (!formsWithErrors.length) return;

  const formsWithErrorsString =
    formsWithErrors.length > 1
      ? formsWithErrors.map(({ name }) => name.toLowerCase()).join(', ')
      : `${formsWithErrors[0].name.toLowerCase()} `;
  const errorMessage = `Add required info in ${formsWithErrorsString}before submitting your data to OEC.`;
  return {
    heading: 'This record has missing or incorrect info',
    type: 'error',
    text: errorMessage,
  };
};

export default EditRecord;
