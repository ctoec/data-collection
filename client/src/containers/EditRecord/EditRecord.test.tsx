import React from 'react';
import EditRecord from './EditRecord';
import { accessibilityTestHelper, snapshotTestHelper } from '../../testHelpers';

jest.mock('../../utils/api');
import * as api from '../../utils/api';
import { waitFor } from '@testing-library/react';
const apiMock = api as jest.Mocked<typeof api>;

const rowData = {
  id: '00000000-0000-0000-0000-000000000000',
  firstName: 'First',
  lastName: 'Last',
  enrollments: [],
  family: {
    id: 1,
    incomeDeterminations: [],
  },
};
describe('EditRecord', () => {
  beforeAll(() =>
    apiMock.apiGet.mockReturnValue(new Promise((resolve) => resolve(rowData)))
  );

  const helperOpts = {
    wrapInRouter: true,
    before: () => waitFor(() => expect(apiMock.apiGet).toBeCalled()),
  };
  snapshotTestHelper(<EditRecord />, helperOpts);
  accessibilityTestHelper(<EditRecord />, helperOpts);

  afterAll(() => jest.clearAllMocks());
});
