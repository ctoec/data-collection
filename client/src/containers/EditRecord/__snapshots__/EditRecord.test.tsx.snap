// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditRecord matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="grid-container"
  >
    <button
      class="usa-button usa-button--unstyled margin-top-2 text-bold"
      type="button"
    >
      <span
        class="oec-text-with-icon"
      >
        <svg
          class="oec-text-with-icon__icon oec-text-with-icon__icon--direction-left oec-text-with-icon__icon--side-left"
        >
          arrowRight.svg
        </svg>
        Back
      </span>
    </button>
    <div
      class="usa-alert usa-alert--error grid-row"
      role="alert"
    >
      <div
        class="display-flex flex-fill"
      >
        <div
          class="usa-alert__body grid-col flex-fill"
        >
          <h2
            class="usa-alert__heading"
          >
            This record has missing or incorrect info
          </h2>
          <p
            class="usa-alert__text"
          >
            Add required info in enrollment and funding before submitting your data to OEC.
          </p>
        </div>
        <div
          class="usa-alert__body flex-auto display-flex flex-align-center"
        />
      </div>
    </div>
    <div
      class="display-flex flex-justify"
    >
      <div>
        <h1
          class="margin-top-0"
        >
          <span
            class="h2 h2--lighter"
          >
            Edit record 
          </span>
          First Last
        </h1>
      </div>
      <div
        class="display-flex flex-col flex-align-center"
      >
        <span
          class="margin-right-2 text-base-light"
        >
          Quick actions
        </span>
        <button
          class="usa-button usa-button--unstyled margin-right-2"
          type="button"
        >
          <span
            class="oec-text-with-icon"
          >
            <svg
              class="oec-text-with-icon__icon oec-text-with-icon__icon--direction-right oec-text-with-icon__icon--side-left"
            >
              change.svg
            </svg>
            Change age group and/or site
          </span>
        </button>
        <button
          class="usa-button usa-button--unstyled margin-right-0"
          type="button"
        >
          <span
            class="oec-text-with-icon"
          >
            <svg
              class="oec-text-with-icon__icon oec-text-with-icon__icon--direction-right oec-text-with-icon__icon--side-left"
              height="20"
              viewBox="0 0 19 20"
              width="19"
              xmlns="http://www.w3.org/2000/svg"
            >
              <title>
                trashCan
              </title>
              <g
                fill="#D63E04"
                fill-rule="nonzero"
              >
                <path
                  d="M3.221 18.8c.063.695.632 1.2 1.327 1.2h10.905c.695 0 1.263-.526 1.327-1.2L18 6H2l1.221 12.8zm9.452-8.8c0-.358.295-.674.674-.674.358 0 .673.295.673.674v6a.678.678 0 0 1-.673.673.678.678 0 0 1-.674-.673v-6zm-3.347 0c0-.358.295-.674.674-.674.358 0 .673.295.673.674v6a.678.678 0 0 1-.673.673.678.678 0 0 1-.674-.673v-6zM6 10c0-.358.294-.674.673-.674.358 0 .674.295.674.674v6a.678.678 0 0 1-.674.673A.678.678 0 0 1 6 16v-6zM17.649 2.15h-5.426v-.725c0-.791-.6-1.425-1.351-1.425H8.149c-.75 0-1.351.634-1.351 1.425v.724H1.35C.601 2.15 0 2.783 0 3.575 0 4.366.6 5 1.351 5H17.65C18.399 5 19 4.366 19 3.575c0-.792-.6-1.426-1.351-1.426z"
                />
              </g>
            </svg>
            Delete record
          </span>
        </button>
      </div>
    </div>
    <div
      class="grid-col"
    >
      <div
        class="oec-loading-wrapper"
      >
        <div
          aria-hidden="false"
          class=""
        >
          <div
            class="oec-tab-nav"
          >
            <div
              class="oec-tab-nav--header"
            >
              <div
                aria-orientation="horizontal"
                class="grid-row oec-tab-nav--tab-container"
                role="tablist"
              >
                <button
                  aria-selected="true"
                  class="oec-tab-nav--tab oec-tab-nav--tab__active"
                  id="identifiers"
                  role="tab"
                  title="Child identifiers"
                  type="button"
                >
                  <span>
                    Child identifiers
                  </span>
                </button>
                <button
                  aria-selected="false"
                  class="oec-tab-nav--tab"
                  id="demographics"
                  role="tab"
                  title="Child info"
                  type="button"
                >
                  <span>
                    Child info
                  </span>
                </button>
                <button
                  aria-selected="false"
                  class="oec-tab-nav--tab"
                  id="family"
                  role="tab"
                  title="Family address"
                  type="button"
                >
                  <span>
                    Family address
                  </span>
                </button>
                <button
                  aria-selected="false"
                  class="oec-tab-nav--tab"
                  id="income"
                  role="tab"
                  title="Family income"
                  type="button"
                >
                  <span>
                    Family income
                  </span>
                </button>
                <button
                  aria-selected="false"
                  class="oec-tab-nav--tab"
                  id="enrollment"
                  role="tab"
                  title="Enrollment and funding"
                  type="button"
                >
                  <span>
                    <span
                      class="oec-text-with-icon svg-gold-20v"
                    >
                      Enrollment and funding
                      <svg
                        class="oec-text-with-icon__icon oec-text-with-icon__icon--direction-right oec-text-with-icon__icon--side-right"
                        height="126"
                        viewBox="0 0 126 126"
                        width="126"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M117.18 31.592c-5.595-9.586-13.185-17.176-22.771-22.771C84.821 3.226 74.354.429 63.001.429c-11.352 0-21.822 2.797-31.408 8.392-9.587 5.594-17.177 13.184-22.772 22.771C3.225 41.179.428 51.649.428 63c0 11.352 2.798 21.82 8.392 31.408 5.595 9.585 13.185 17.176 22.772 22.771 9.587 5.595 20.056 8.392 31.408 8.392s21.822-2.797 31.408-8.392c9.586-5.594 17.176-13.185 22.771-22.771 5.594-9.587 8.391-20.057 8.391-31.408 0-11.352-2.797-21.822-8.39-31.408zM52.572 16.071c0-.761.244-1.385.733-1.874.489-.488 1.114-.733 1.874-.733h15.644c.76 0 1.385.245 1.872.733.488.489.734 1.113.734 1.874v13.036c0 .76-.246 1.385-.734 1.873-.487.489-1.112.733-1.872.733H55.179c-.76 0-1.385-.244-1.874-.733-.489-.488-.733-1.113-.733-1.873V16.071zm31.285 86.036c0 .76-.246 1.385-.733 1.872-.487.489-1.112.733-1.874.733h-36.5c-.761 0-1.385-.244-1.874-.733-.489-.488-.733-1.113-.733-1.873V89.07c0-.762.244-1.385.733-1.874.489-.488 1.114-.733 1.874-.733h7.822V60.392H44.75c-.761 0-1.385-.244-1.874-.733-.489-.488-.733-1.113-.733-1.874V44.75c0-.761.244-1.385.733-1.874.489-.489 1.114-.733 1.874-.733h26.073c.76 0 1.385.244 1.872.733.488.489.734 1.113.734 1.874v41.714h7.82c.761 0 1.386.245 1.874.733.487.488.733 1.113.733 1.874v13.036h.001z"
                          fill="#1B1B1B"
                        />
                      </svg>
                    </span>
                  </span>
                </button>
                <div
                  class="oec-tab-nav--dropdown opacity-0"
                >
                  <button
                    aria-expanded="false"
                    aria-haspopup="false"
                    aria-hidden="true"
                    class="oec-tab-nav--tab"
                    disabled=""
                  >
                    +0 tabs
                  </button>
                </div>
              </div>
            </div>
            <div
              aria-labelledby="identifiers"
              class="oec-tab-nav--content"
              role="tabpanel"
            >
              <form
                autocomplete="off"
                class="usa-form ChildIdentifiersForm usa-form"
                novalidate=""
              >
                <h2>
                  Child's identifiers
                </h2>
                <div
                  class="mobile-lg:grid-col-9"
                >
                  <div
                    class=" usa-form-group"
                  >
                    <label
                      class="usa-label"
                      for="firstName"
                    >
                      First name 
                    </label>
                    <input
                      aria-required="true"
                      class="usa-input oec-text-input"
                      id="firstName"
                      type="text"
                      value="First"
                    />
                  </div>
                </div>
                <div
                  class="mobile-lg:grid-col-9"
                >
                  <div
                    class=" usa-form-group"
                  >
                    <label
                      class="usa-label"
                      for="middleName"
                    >
                      Middle name 
                      <span
                        class="usa-hint"
                      >
                        Â (optional)
                      </span>
                    </label>
                    <input
                      aria-required="false"
                      class="usa-input oec-text-input"
                      id="middleName"
                      type="text"
                      value=""
                    />
                  </div>
                </div>
                <div
                  class="display-flex flex-row flex-align-end grid-row grid-gap"
                >
                  <div
                    class="mobile-lg:grid-col-9"
                  >
                    <div
                      class=" usa-form-group"
                    >
                      <label
                        class="usa-label"
                        for="lastName"
                      >
                        Last name 
                      </label>
                      <input
                        aria-required="true"
                        class="usa-input oec-text-input"
                        id="lastName"
                        type="text"
                        value="Last"
                      />
                    </div>
                  </div>
                  <div
                    class="mobile-lg:grid-col-3"
                  >
                    <div
                      class=" usa-form-group"
                    >
                      <label
                        class="usa-label"
                        for="suffix"
                      >
                        Suffix 
                        <span
                          class="usa-hint"
                        >
                          Â (optional)
                        </span>
                      </label>
                      <input
                        aria-required="false"
                        class="usa-input oec-text-input"
                        id="suffix"
                        type="text"
                        value=""
                      />
                    </div>
                  </div>
                </div>
                <fieldset
                  aria-labelledby="fieldset-legend-dateOfBirth-picker dateOfBirth-picker-hint"
                  class="usa-fieldset"
                  id="dateOfBirth-picker"
                >
                  <legend
                    id="fieldset-legend-dateOfBirth-picker"
                  >
                    <span
                      class="usa-label"
                    >
                      Birthdate
                    </span>
                  </legend>
                  <span
                    class="usa-hint text-italic"
                    id="dateOfBirth-picker-hint"
                  >
                    For example: 6 21 2008
                  </span>
                  <div>
                    <div
                      class="flex-row flex-align-end usa-memorable-date"
                    >
                      <div
                        class="oec-date-input__input margin-right-1 margin-top-0 usa-form-group"
                      >
                        <label
                          class="usa-label"
                          for="dateOfBirth-picker-month"
                        >
                          Month 
                        </label>
                        <input
                          aria-required="true"
                          class="usa-input oec-text-input usa-input--inline"
                          id="dateOfBirth-picker-month"
                          max="12"
                          min="1"
                          type="number"
                          value=""
                        />
                      </div>
                      <div
                        class="oec-date-input__input margin-right-1 margin-top-0 usa-form-group"
                      >
                        <label
                          class="usa-label"
                          for="dateOfBirth-picker-day"
                        >
                          Day 
                        </label>
                        <input
                          aria-required="true"
                          class="usa-input oec-text-input usa-input--inline"
                          id="dateOfBirth-picker-day"
                          max="31"
                          min="1"
                          type="number"
                          value=""
                        />
                      </div>
                      <div
                        class="oec-date-input__input margin-right-1 margin-top-0 usa-form-group"
                      >
                        <label
                          class="usa-label"
                          for="dateOfBirth-picker-year"
                        >
                          Year 
                        </label>
                        <input
                          aria-required="true"
                          class="usa-input oec-text-input usa-input--inline"
                          id="dateOfBirth-picker-year"
                          max="2200"
                          min="1000"
                          type="number"
                          value=""
                        />
                      </div>
                    </div>
                  </div>
                </fieldset>
                <fieldset
                  aria-labelledby="fieldset-legend-birth-certificate-fields"
                  class="usa-fieldset"
                  id="birth-certificate-fields"
                >
                  <legend
                    id="fieldset-legend-birth-certificate-fields"
                  >
                    <span
                      class="usa-label"
                    >
                      Birth certificate
                    </span>
                  </legend>
                  <div>
                    <span>
                      <div
                        class="usa-radio"
                      >
                        <input
                          class="usa-radio__input"
                          id="US-birth-certificate"
                          name="birthCertificateType"
                          type="radio"
                          value="US birth certificate"
                        />
                        <label
                          class="usa-radio__label"
                          for="US-birth-certificate"
                        >
                          US birth certificate
                        </label>
                      </div>
                    </span>
                    <span>
                      <div
                        class="usa-radio"
                      >
                        <input
                          class="usa-radio__input"
                          id="Non-US-birth-certificate"
                          name="birthCertificateType"
                          type="radio"
                          value="Non-US birth certificate"
                        />
                        <label
                          class="usa-radio__label"
                          for="Non-US-birth-certificate"
                        >
                          Non-US birth certificate
                        </label>
                      </div>
                    </span>
                    <span>
                      <div
                        class="usa-radio"
                      >
                        <input
                          checked=""
                          class="usa-radio__input"
                          id="Unavailable"
                          name="birthCertificateType"
                          type="radio"
                          value="Unavailable"
                        />
                        <label
                          class="usa-radio__label"
                          for="Unavailable"
                        >
                          Unavailable
                        </label>
                      </div>
                    </span>
                  </div>
                </fieldset>
                <input
                  class="usa-button"
                  type="submit"
                  value="Save"
                />
              </form>
            </div>
          </div>
        </div>
        <div
          class="oec-loading-wrapper__indicator bg-white"
          hidden=""
        >
          <div
            class="MuiCircularProgress-root indeterminate MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
            role="progressbar"
            style="width: 40px; height: 40px;"
          >
            <svg
              class="MuiCircularProgress-svg"
              viewBox="22 22 44 44"
            >
              <circle
                class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
                cx="44"
                cy="44"
                fill="none"
                r="20.2"
                stroke-width="3.6"
              />
            </svg>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
